src/main/java/com/rapid7/container/analyzer/docker/packages/OwaspDependencyParser.java

```diff
@@ -58,9 +58,14 @@ public boolean supports(String name, TarArchiveEntry entry) {
   }
 
   private Package convertDependencyToPackage(Dependency dependency) {
+
+    if (dependency.getName() == null && "Java".equalsIgnoreCase(dependency.getEcosystem())) {
+      dependency = Spring4ShellParser.getPackageNameandVersion(dependency);
+    }
     if (dependency.getName() == null || dependency.getVersion() == null || dependency.getEcosystem() == null) {
       return null;
     }
```

Comment:
This isn't going to work. If name is null, it will first set dependency using Spring4ShellParser, and it will then fall through to line 66 and return null. You can make it an else if statement to get around that problem